go-controller/pkg/ovn/service.go

```diff
@@ -141,6 +155,13 @@ func (ovn *Controller) syncServices(services []interface{}) {
 						"loadbalancer %s", vip, loadBalancer)
 					ovn.deleteLoadBalancerVIP(loadBalancer, vip)
 				}
+
+				// handle external IP based vips directly
+				if !stringSliceMembership(lbServices[protocol], vip) {
```

Comment:
This looks wrong to me too. I think we will end up deleting all nodeport VIPs with this.

Reply:
I think it still has the problem. If VM IP is 10.0.0.10 and extenralIP is 20.0.0.10, then since 10.0.0.10:1234 is not in lbServices, you will delete 10.0.0.10:1234